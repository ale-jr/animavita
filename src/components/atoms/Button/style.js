import styled from "styled-components";

export const Button = styled.button`
  background-color: ${({ theme }) => theme.colors.primary.medium};
  color: ${({ theme }) => theme.colors.primary.contrast};
  font-weight: ${({ theme }) => theme.typography.weights.bold};
  border-radius: 500px;
  padding: ${({ theme }) => `${theme.spacing.xxs} ${theme.spacing.xs}`};
  text-transform: uppercase;
  border: none;
  cursor: pointer;

  transition: ${({ theme }) => theme.animations.easeInOut};

  &:hover {
    background-color: ${({ theme }) => theme.colors.primary.dark};
  }

  &:focus-visible {
    background-color: ${({ theme }) => theme.colors.primary.medium};
    outline-color: ${({ theme }) => theme.colors.primary.dark};
  }
`;
