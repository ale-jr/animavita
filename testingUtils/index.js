import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../src/themes/default";

function ThemeProviderWrapper({ children }) {
  return (
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  );
}

export const renderWithTheme = (ui, options) =>
  render(ui, {
    wrapper: ThemeProviderWrapper,
    ...options
  });
