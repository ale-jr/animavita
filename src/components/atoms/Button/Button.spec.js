import Button from ".";
import { renderWithTheme } from "../../../../testingUtils";
import { fireEvent } from "@testing-library/react";
describe("Button", () => {
  it("Matches snapshot", () => {
    const { container } = renderWithTheme(<Button>Click me!</Button>);
    expect(container).toMatchSnapshot();
  });

  it("Fires click event", () => {
    const onClick = jest.fn();
    const { container } = renderWithTheme(
      <Button onClick={onClick}>Click me!</Button>
    );
    fireEvent.click(container.firstChild);

    expect(onClick).toHaveBeenCalled();
  });
});
