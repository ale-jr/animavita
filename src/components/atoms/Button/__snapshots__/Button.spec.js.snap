// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Matches snapshot 1`] = `
<div>
  <button
    class="sc-bdvvtL bxDyPj"
  >
    Click me!
  </button>
</div>
`;
